packages/ra-ui-materialui/src/input/SelectArrayInput.js

```diff
@@ -201,13 +201,13 @@ export class SelectArrayInput extends Component {
                                 .filter(choice =>
                                     selected.includes(get(choice, optionValue))
                                 )
-                                .map(choice => (
-                                    <Chip
-                                        key={get(choice, optionValue)}
-                                        label={get(choice, optionText)}
-                                        className={classes.chip}
-                                    />
-                                ))}
+                                .map(choice =>
+                                    React.cloneElement(children, {
+                                        key: get(choice, optionValue),
+                                        record: choice,
+                                        className: classes.chip,
```

Comment:
I think we need to pass more props to the child. The `ChipField`, for instance, expects a `source` in addition to the `record`.

Please add a unit test to prove that your change works.

Reply:
No, the `source` prop is setup in the component, so already available. For example: 
```
<SelectArrayInput>
    <ChipField source="name"/>
</SelectArrayInput>
```

Reply:
Test added

Reply:
you're right, my bad